text/0000-std-function-macro.md

```diff
@@ -0,0 +1,59 @@
+- Feature Name: Function Name Macro
+- Start Date: (2016-08-19)
+- RFC PR: (leave this empty)
+- Rust Issue: (leave this empty)
+
+# Summary
+[summary]: #summary
+
+`function` is a macro, which expands to the fully qualified function name from which
+it was invoked. It works similar to `std::file` or `std::line` and expands to
+an expression of type `&'static str`.
+
+# Motivation
+[motivation]: #motivation
+
+There are currently macros for getting the source file name, `std::file`, and
+the line number, `std::line`. To complete this pair, a macro to for getting
+the current function name would be useful. All three macros are useful for
+e.g. generating log output with more context about where something was
+happening. The file and line is good for being able to look up the exact
+location, however the fully qualified function name gives additional context
+without even having to look into the source code.
+
+# Detailed design
+[design]: #detailed-design
+
+This feature would have an equivalent design and implementation to
+`std::file`. It would return the fully qualified function name, which would be
+the same as seen with `RUST_BACKTRACE=1`, e.g.
+
+ - `hello::main::h14dbb225e6ef2d49` for function `main` crate `hello`
```

Comment:
This seems... less than good? Why are we trying to give this information. I'd *much* rather know just the fully qualified function name:

* `hello::main`
* `hello::bar::foo`
* `hello::bar::Foo::new`
* `hello::bar::Foo<T>::new`